testsuite/MDAnalysisTests/coordinates/test_xdr.py

```diff
@@ -25,257 +25,209 @@
+        universe.trajectory.rewind()
+        uc = universe.coord.dimensions
+        assert_almost_equal(
             uc,
             self.ref_unitcell,
             self.prec,
             err_msg="unit cell dimensions (rhombic dodecahedron)")
 
-
-    def test_volume(self):
+    def test_volume(self, universe):
         # need to reduce precision for test (nm**3 <--> A**3)
-        self.universe.trajectory.rewind()
-        vol = self.ts.volume
-        assert_array_almost_equal(
+        universe.trajectory.rewind()
+        vol = universe.coord.volume
+        assert_almost_equal(
             vol,
             self.ref_volume,
             0,
             err_msg="unit cell volume (rhombic dodecahedron)")
 
+    def test_dt(self, universe):
+        assert_almost_equal(
+            universe.trajectory.dt, 100.0, 4, err_msg="wrong timestep dt")
 
-    def test_dt(self):
-        assert_almost_equal(self.universe.trajectory.dt,
-                            100.0,
-                            4,
-                            err_msg="wrong timestep dt")
-
-
-    def test_totaltime(self):
+    def test_totaltime(self, universe):
         # test_totaltime(): need to reduce precision because dt is only precise
         # to ~4 decimals and accumulating the inaccuracy leads to even lower
         # precision in the totaltime (consequence of fixing Issue 64)
-        assert_almost_equal(self.universe.trajectory.totaltime,
-                            900.0,
-                            3,
-                            err_msg="wrong total length of trajectory")
-
-
-    def test_frame(self):
-        self.trajectory[4]  # index is 0-based and frames are 0-based
-        assert_equal(self.universe.trajectory.frame, 4, "wrong frame number")
-
+        assert_almost_equal(
+            universe.trajectory.totaltime,
+            900.0,
+            3,
+            err_msg="wrong total length of trajectory")
 
-    def test_time(self):
-        self.trajectory[4]
-        assert_almost_equal(self.universe.trajectory.time,
-                            400.0,
-                            3,
-                            err_msg="wrong time of frame")
+    def test_frame(self, universe):
+        universe.trajectory[4]  # index is 0-based and frames are 0-based
+        assert_equal(universe.trajectory.frame, 4, "wrong frame number")
 
+    def test_time(self, universe):
+        universe.trajectory[4]
+        assert_almost_equal(
+            universe.trajectory.time, 400.0, 3, err_msg="wrong time of frame")
 
-    def test_get_Writer(self):
-        W = self.universe.trajectory.Writer(self.outfile)
-        assert_equal(self.universe.trajectory.format, W.format)
-        assert_equal(self.universe.atoms.n_atoms, W.n_atoms)
+    def test_get_Writer(self, universe, tmpdir):
+        ext = os.path.splitext(self.filename)[1]
+        outfile = str(tmpdir.join('/xdr-reader-test' + ext))
+        W = universe.trajectory.Writer(outfile)
```

Comment:
Please use 
```python
with universe.trajectory.Writer(outfile) as W:
   ...
```
instead of the explicit `W.close()`. The context manager is safer and also the "best practice".

Reply:
I was simply to lazy to change all the occurrences.